yarpcerrors/errors.go

```diff
@@ -25,24 +25,112 @@ import (
 	"fmt"
 )
 
-// IsYARPCError is a convenience function that returns true if the given error
-// is a non-nil YARPC error.
-func IsYARPCError(err error) bool {
-	_, ok := err.(*yarpcError)
+// Newf returns a new Status.
+//
+// The Code should never be CodeOK, if it is, this will return nil.
+func Newf(code Code, format string, args ...interface{}) *Status {
+	if code == CodeOK {
+		return nil
+	}
+	return &Status{
+		code:    code,
+		message: sprintf(format, args...),
+	}
+}
+
+// FromError returns the Status for the error.
+//
+// If the error is a Status, it returns err.
+// If the error is not a Status, it returns a new error with the Code
+// CodeUnknown and the message err.Error()
+func FromError(err error) *Status {
```

Comment:
add tests plz

Reply:
Yea, going go through through everything and update it